src/lib/comms/sol-oic-server.c

```diff
@@ -183,12 +183,41 @@ static const struct sol_oic_resource_type oic_p_resource_type = {
     }
 };
 
+static bool
+iface_contains(const char *iface, struct sol_str_slice query)
+{
+
+    const char *p = NULL;
+    struct sol_str_slice token, slice = sol_str_slice_from_str(iface),
+        delim = SOL_STR_SLICE_LITERAL(" ");
+
+    while (sol_str_slice_split_iterate(slice, &token, &p, delim)) {
+        if (sol_str_slice_eq(query, token))
+            return true;
+    }
+
+    return false;
+}
+
+static bool
+oic_query_split(struct sol_str_slice query, struct sol_str_slice *key, struct sol_str_slice *value)
+{
+    char *sep;
+
+    sep = memchr(query.data, '=', query.len);
+    SOL_NULL_CHECK(sep, false);
+    *key = SOL_STR_SLICE_STR(query.data, sep - query.data);
+    *value = SOL_STR_SLICE_STR(sep + 1, query.len - (sep - query.data + 1));
```

Comment:
Cool. So please make `sep` const. (I really don't like these libc functions that drop the const)